src/components/legend/draw.js

```diff
@@ -415,48 +460,21 @@ function computeLegendDimensions(gd, traces) {
     opts.height = 0;
 
     traces.each(function(d) {
-        var trace = d[0].trace,
-            g = d3.select(this),
-            bg = g.selectAll('.legendtoggle'),
-            text = g.selectAll('.legendtext'),
-            tspans = g.selectAll('.legendtext>tspan'),
-            tHeight = opts.font.size * 1.3,
-            tLines = tspans[0].length || 1,
-            tWidth = text.node() && Drawing.bBox(text.node()).width,
-            mathjaxGroup = g.select('g[class*=math-group]'),
-            textY,
-            tHeightFull;
-
-        if(!trace.showlegend) {
-            g.remove();
-            return;
-        }
+        var legendItem = d[0];
 
-        if(mathjaxGroup.node()) {
-            var mathjaxBB = Drawing.bBox(mathjaxGroup.node());
-            tHeight = mathjaxBB.height;
-            tWidth = mathjaxBB.width;
-            mathjaxGroup.attr('transform','translate(0,' + (tHeight / 4) + ')');
-        }
-        else {
-            // approximation to height offset to center the font
-            // to avoid getBoundingClientRect
-            textY = tHeight * (0.3 + (1 - tLines) / 2);
-            text.attr('y', textY);
-            tspans.attr('y', textY);
-        }
+        computeTextDimensions(gd, this, legendItem);
```

Comment:
I would prefer having:

```js
var container = d3.select(this);

/* ... */

container.call(computeTextDimensions, gd, legendItem);
```

than passing `this` to `computeTextDimensions`, which can be pretty confusing.